src/App.jsx

```diff
@@ -1,14 +1,36 @@
 import "./App.css";
-import { Fragment } from "react";
-import useSemiPersistentState from "./hooks/useSemiPersistentState.jsx";
+import { Fragment, useState, useEffect } from "react";
 import TodoList from "./components/TodoList.jsx";
 import AddTodoForm from "./components/AddTodoForm.jsx";
 
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState(
-    "savedTodoList",
-    JSON.parse(localStorage.getItem("savedTodoList")) || []
-  );
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    new Promise((resolve, reject) => {
+      setTimeout(
+        () =>
+          resolve({
+            data: {
+              todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
+            },
+          }),
+        2000
+      );
+    })
+      .then((result) => {
+        setIsLoading(true);
+        setTodoList(result.data.todoList);
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+    }
+  }, [todoList, isLoading]);
```

Comment:
You need to remove the dependency on isLoading because you only need to save todoList to localStorage after it changes, not when the loading state is toggled.